packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -702,28 +700,27 @@ export class CallContext {
     return id;
   }
 
-  private processNewCaption(captions: CaptionInfoState[], captionInfo: CaptionInfoState): void {
+  private processNewCaption(captions: CaptionsInfo[], newCaption: CaptionsInfo): void {
     // Check if the incoming caption is a replacement for the latest one
     const latestCaption = captions[captions.length - 1];
     if (
       latestCaption &&
-      latestCaption.resultType !== ResultType.Final &&
-      latestCaption.timestamp.getTime() === captionInfo.timestamp.getTime() &&
-      toFlatCommunicationIdentifier(latestCaption.speaker.identifier) ===
-        toFlatCommunicationIdentifier(captionInfo.speaker.identifier)
+      latestCaption.resultType !== 'Final' &&
+      latestCaption.timestamp.getTime() === newCaption.timestamp.getTime() &&
+      latestCaption.speaker.identifier === newCaption.speaker.identifier
```

Comment:
These 2 are objects, which we cannot directly compare (which is comparing ref),
Do
```
toFlatCommunicationIdentifier(latestCaption.speaker.identifier) ===
        toFlatCommunicationIdentifier(newCaption.speaker.identifier)
```
Instead

Reply:
Fixed